### PR TITLE
Refactor/separate methods in classes

### PR DESCRIPTION
**Pull request by jExbrayat**
Split the single class detrend in detrend.py into several classes.

Create _fancy_plot function in detrend_fancy_plot.py.
It is meant to be called in the fancy_plot method in each detrend class.

### CODE REVIEW

File: src/detrend_fancy_plot.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+def _fancy_plot(
+    y_original: np.ndarray | pd.DataFrame,
+    y_fitted: np.ndarray,
+    y_detrend: np.ndarray,
+    fitted_parameters: dict,
+    xticklabels: pd.core.indexes.base.Index | None = None,
+    method_name: str = "",
+) -> None:
+    """plot two graphs : the original data and its fitted trend curve ; the detrended data
```
Comment (Ab2nour) [inline src/detrend_fancy_plot.py]: nit: "Plot two graphs: the original [...]"

=> uppercase letter at the start of docstring
=> no space before ":"
=> also, no space before ";"

(nit means nitpicking => le mec qui clc quoi, et du coup tu préfixes de "nit" pour dire que tu attaques sur des détails, pour ne pas contrarier l'autre. Car il faut quand même pointer tous les petits détails, ça reste important)

File: src/detrend.py [hunk 2]
```diff
@@ -102,39 +112,135 @@ def _PolynomialRegression(self, y: np.ndarray | pd.DataFrame) -> np.ndarray:
...
+    def __init__(self, window: int = 100) -> None:
+        self.fitted_parameters = {"Time span": window}
+        self.window = window
+        self.method_name = "linear mobile average"
 
-    def _LinearMA(self, y: np.ndarray | pd.DataFrame) -> np.ndarray:
+    def fit(self, y: np.ndarray | pd.DataFrame) -> np.ndarray:
         """
-        returns fitted values with the linear centered mobile average method
+        returns fitted values with the linear mobile average method
```
Comment (Ab2nour) [inline src/detrend.py]: nit: uppercase at the start of docstring "Returns fitted..."

File: src/detrend.py [hunk 3]
```diff
@@ -102,39 +112,135 @@ def _PolynomialRegression(self, y: np.ndarray | pd.DataFrame) -> np.ndarray:
...
+        """_summary_
+
+        Returns:
+            np.ndarray: detrended values, 1 dimensional array of length len(y)
+        """
+        self.y_predict = self.y_original - self.fitted_values
+        return self.y_predict
+
+    def fancy_plot(self, xticklabels: pd.core.indexes.base.Index | None = None) -> None:
+        """plot two graphs : the original data and its fitted trend curve ; the detrended data
```
Comment (Ab2nour) [inline src/detrend.py]: nit: uppercase at start of docstring + no space before ":" and ";"

File: src/detrend.py [hunk 4]
```diff
@@ -102,39 +112,135 @@ def _PolynomialRegression(self, y: np.ndarray | pd.DataFrame) -> np.ndarray:
...
+
+        Returns:
+            np.ndarray: detrended values, 1 dimensional array of length len(y)
         """
-        returns fitted values with the BSplines interpolation method
+        self.y_predict = self.y_original - self.fitted_values
+        return self.y_predict
+
+    def fancy_plot(self, xticklabels: pd.core.indexes.base.Index | None = None) -> None:
+        """plot two graphs : the original data and its fitted trend curve ; the detrended data
```
Comment (Ab2nour) [inline src/detrend.py]: nit: same

File: src/detrend.py [hunk 5]
```diff
@@ -102,39 +112,135 @@ def _PolynomialRegression(self, y: np.ndarray | pd.DataFrame) -> np.ndarray:
...
             "Smoothing factor": smoothing_factor,
             "Degree": degree,
         }
+        self.smoothing_factor = smoothing_factor
+        self.degree = degree
+        self.method_name = "B-splines"
 
+    def fit(self, y: np.ndarray | pd.DataFrame) -> np.ndarray:
+        """
+        returns fitted values with the B-splines method
```
Comment (Ab2nour) [inline src/detrend.py]: nit: "Returns"

**Review by Ab2nour — CHANGES_REQUESTED** [review]
Only nitpicking. Otherwise, it's all good

File: src/detrend_fancy_plot.py (hunk 1 above)
Comment (jExbrayat) [reply src/detrend_fancy_plot.py]: done boi

Comment (Ab2nour) [reply src/detrend_fancy_plot.py]: you forgot to commit boi

File: src/detrend.py (hunk 2 above)
Comment (jExbrayat) [reply src/detrend.py]: done

File: src/detrend.py (hunk 4 above)
Comment (jExbrayat) [reply src/detrend.py]: done

File: src/detrend.py (hunk 5 above)
Comment (jExbrayat) [reply src/detrend.py]: done

**Comment by Ab2nour** [comment]
LGTM!

**Pull request merged**